### PR TITLE
Fixing bug with approval form

### PR DESCRIPTION
**Pull request by rossneilson**
## Description
Currently, any submission in the linked button is triggering the onSubmit of the form. 
Previously this was not an issue since there was only one button. Having the onSubmit form is nice for accessibility and using keyboard to submit so the solution is just a check to ensure approvalState is approved before purchasing tokens.
Also stopped spinner if an error occurs with approvals.

## Motivation and Context
#297 

## How Has This Been Tested?
Locally on new xdai sale

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by mprasanjith** [comment]
The PR looks good, but can't we use `type="button"` and `type="submit"` to take control of this?

That way when approval is pending we can set the Approve button to `type="submit"`, and later switch it once we have the approval. Just an accessibility and UX nice-to-have.

**Comment by rossneilson** [comment]
Nice idea, is the latest commit what you meant?

**Review by mprasanjith — APPROVED** [review]
Yes, it is. Additionally, I was thinking that we can set typeSubmit dynamically, e.g.:
1. When the purchase hasn't been approved, we can make the Approve button to be `typeSubmit={true}` and Purchase button to be `typeSubmit={false}`.
2. When the purchase is approved, we can make the Purchase button to be `typeSubmit={true}` and Approve button to be `typeSubmit={false}`.

It's not so important though. I'm approving the PR as this isn't really a necessity.

**Comment by rossneilson** [comment]
Ah, I see what you mean. Since it would require a few more changes I'll maybe leave it just now. It would also limit the flexibility of the component in a way. But it might be good to add in future.

**Pull request merged**
